### PR TITLE
Adds missing content-type header to api docs

### LINKED ISSUE #1450: Mention the inclusion of api headers in API docs
## Abstract

As a user found out in TG, calling the API without including the content-type header leads to a failure. 

```curl
curl 'http://localhost:4243/api/1/blockchains/ETH' \
  -X 'PUT' \
  -H 'Accept: application/json, text/plain, */*' \
  -H 'Content-Type: application/json;charset=UTF-8' \
  --data-raw '{"accounts":[{"address":"0x","label":"","tags":[]}]}
```

This works  removing the contentType header seems to give a 400 error

## Task

Add it to the docs tha tthis heaer is a requirement for every call that it is.

### PR DESCRIPTION
**Pull request by kelsos**
Closes #1450

The header has been added to any request regardless of the verb in case it has a JSON body payload.

### CODE REVIEW

**Comment by kelsos** [comment]
Interesting

**Review by LefterisJP — COMMENTED** [review]
Something is happening with the addition ofthe Content-Type header. You would have to check the logs to see where json parsing fails.

**Comment by kelsos** [comment]
I think I have an idea. Some of the "json" examples are not valid json... without the content type they were not validated but now they are :laughing:

**Comment by LefterisJP** [comment]
> I think I have an idea. Some of the "json" examples are not valid json... without the content type they were not validated but now they are

yeah that's strange. Cause some of them were validated before and we had to fix them. Dunno. But I also put some print statements and seems some commas are not needed, or missing and minor stuff like that

**Comment by kelsos** [comment]
> > I think I have an idea. Some of the "json" examples are not valid json... without the content type they were not validated but now they are
> 
> yeah that's strange. Cause some of them were validated before and we had to fix them. Dunno. But I also put some print statements and seems some commas are not needed, or missing and minor stuff like that

Yes can you give me some idea of what to edit and where to put the print statements?

**Comment by LefterisJP** [comment]
To see the json as the httpexamples addon parses it:

Open `/home/lefteris/.virtualenvs/rotkipy37/lib/python3.7/site-packages/sphinxcontrib/httpexample/parsers.py`   (location varies on your venv)

Add: `print(payload_str)` between lines 116 and 117

File: docs/api.rst [hunk 1]
```diff
@@ -2270,18 +2270,19 @@ Editing custom ethereum tokens
...
-          "address": "0x1169C72f36A843cD3a3713a76019FAB9503B2807",
-          "decimals": 5,
-          "name": "foo",
-          "symbol": "FTK",
-          "started": 1614636432,
-          "swapped_for": "SCP,
-          "coingecko": "foo-coin",
-          "cryptocompare": "FOO",
-          "protocol": "aave",
-          "underlying_tokens": None
```
Comment (LefterisJP) [inline docs/api.rst:2255]: wasn't the problem here that it was `None` and not `null`?

**Review by LefterisJP — APPROVED** [review]
Thanks for making the json valid again

File: docs/api.rst (hunk 1 above)
Comment (kelsos) [reply docs/api.rst:2255]: Yes but then I changed it to null. And null was giving me this exception that I linked in discord

```
  File "/home/kelsos/.virtualenvs/rotki/lib/python3.7/site-packages/sphinxcontrib/httpexample/builders.py", line 213, in astify_json_obj
    raise Exception('Cannot astify {0:s}'.format(str(obj)))
Exception: Cannot astify None
```

Comment (kelsos) [reply docs/api.rst:2255]: Yes but then I changed it to null. And null was giving me this exception that I linked in discord

```
  File "/home/kelsos/.virtualenvs/rotki/lib/python3.7/site-packages/sphinxcontrib/httpexample/builders.py", line 213, in astify_json_obj
    raise Exception('Cannot astify {0:s}'.format(str(obj)))
Exception: Cannot astify None
```

**Pull request merged**